main/main.cpp

```diff
@@ -1650,6 +1658,8 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 	GLOBAL_DEF("debug/settings/stdout/print_gpu_profile", false);
 	GLOBAL_DEF("debug/settings/stdout/verbose_stdout", false);
 
+	Engine::get_singleton()->set_session_id(init_session_id);
```

Comment:
The session ID (or index) may not equal to the index of RemoteDebugger because of the execution order of connections.